_search-plugins/concurrent-segment-search.md

```diff
@@ -156,7 +150,25 @@ Additionally, some [Profile API]({{site.url}}{{site.baseurl}}/api-reference/prof
 
 ## Limitations
 
-Parent aggregations on [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) fields do not support the concurrent search model. Thus, if a search request contains a parent aggregation, the aggregation will be executed using the non-concurrent path even if concurrent segment search is enabled at the cluster level.
+The following aggregations do not support the concurrent search model. If a search request contains one of these aggregations, the request will be executed using the non-concurrent path even if concurrent segment search is enabled at the cluster level or index level.
+- Parent aggregations on [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) fields
+- Sampler and Diversified Sampler aggregations.
+
+## Things to know
+
+### The `terminate_after` search parameter
+
+The [`terminate_after` search parameter]({{site.url}}{{site.baseurl}}/api-reference/search/#url-parameters) is used to terminate a search request once a specified number of documents has been collected. If you include the `terminate_after` parameter in a request, concurrent segment search is disabled and the request is run in a non-concurrent manner.
+
+Typically, queries are used with smaller `terminate_after` values and thus complete quickly because the search is performed on a reduced dataset. Therefore, concurrent search may not further improve performance in this case. Moreover, when `terminate_after` is used with other search request parameters, such as `track_total_hits` or `size`, it adds complexity and changes the expected query behavior. Falling back to a non-concurrent path for search requests that include `terminate_after` ensures consistent results between concurrent and non-concurrent requests.
+
+### Sorting
+
+When performing sort operations, optimizations from Lucene that allow for pruning entire segments can greatly speed up the request. Whether these optimizations are able to kick in depends on both the segment layout and the query construction. For concurrent segment search, segments are grouped into segment slices, so in certain cases these sort optimizations may not be applicable to the segment slices. In these cases, you may see an increase in query latency when performing sort operations.
+
+### Terms aggregations
+
+When performing concurrent segment search, the `shard_size` parameter will be applied at the segment slice level. This may introduce additional document count error, which is correctly calculated by the `doc_count_error_upper_bound` response parameter for such cases.
```

Comment:
Do we need to add the note about cases where result may vary w.r.t concurrent search disabled depending on how shard_size value is selected and applied at slice levels? Or is that captured in generic doc section? Basically same issue w.r.t coordinator to shards will be now seen w.r.t shards to slices.

Will be good to add reference to issue or PR where you have details about these cases.

Reply:
I think the concept of document count error generically is covered by https://github.com/opensearch-project/documentation-website/pull/6355 but I think we can link to https://github.com/opensearch-project/OpenSearch/issues/11680#issuecomment-1885882985 as well to cover how the specifically affects concurrent search.